Source: written by David Goulet
Upstream: yes
Reason: This patch avoids a crash in KTP where opdata is dereferenced

From d748757e581b74e7298df155ad49174cb914102b Mon Sep 17 00:00:00 2001
From: David Goulet <dgoulet@ev0ke.net>
Date: Sat, 19 Jan 2013 15:53:23 -0500
Subject: [PATCH] Fix: pass opdata when sending message fragment

The inject_message callback was missing the opdata when sending message
fragments.

Signed-off-by: David Goulet <dgoulet@ev0ke.net>
---
 src/message.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/message.c b/src/message.c
index 6b842bc..fe6957c 100644
--- a/src/message.c
+++ b/src/message.c
@@ -441,7 +441,7 @@ fragment:
 	/* Fragment and send according to policy */
 	if (!err && messagep && *messagep) {
 	    if (context) {
-		err = fragment_and_send(ops, NULL, context, *messagep,
+		err = fragment_and_send(ops, opdata, context, *messagep,
 			fragPolicy, messagep);
 	    }
 	}
-- 
2.1.0

